File: app/src/main/java/io/jeffchang/buttondemo/ui/MainActivity.java
package io.jeffchang.buttondemo.ui;

import android.os.Bundle;
import android.support.annotation.Nullable;

import dagger.android.support.DaggerAppCompatActivity;
import io.jeffchang.buttondemo.R;
import io.jeffchang.buttondemo.ui.usercreate.view.DogDetailFragment;
import io.jeffchang.buttondemo.ui.userlist.view.UserListFragment;

/**
 * Hosts our two main fragments.
 */

public class MainActivity extends DaggerAppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        goToListFragment();
    }

    void goToDetailFragment() {
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.main_fragment_container, DogDetailFragment.newInstance())
                .commit();
    }

    void goToListFragment() {
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.main_fragment_container, UserListFragment.newInstance())
                .commit();
    }
}

File: app/src/main/java/io/jeffchang/buttondemo/ui/MainActivity.kt
package io.jeffchang.buttondemo.ui

import android.os.Bundle

import dagger.android.support.DaggerAppCompatActivity
import io.jeffchang.buttondemo.R
import io.jeffchang.buttondemo.ui.usercreate.view.UserCreateFragment
import io.jeffchang.buttondemo.ui.userlist.view.UserListFragment

/**
 * Hosts our two main fragments.
 */

class MainActivity : DaggerAppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        goToListFragment()
    }

    private fun goToDetailFragment() {
        supportFragmentManager
                .beginTransaction()
                .replace(R.id.main_fragment_container, UserCreateFragment.newInstance())
                .commit()
    }

    private fun goToListFragment() {
        supportFragmentManager
                .beginTransaction()
                .replace(R.id.main_fragment_container, UserListFragment.newInstance())
                .commit()
    }
}
